### PR TITLE
Adding a test case for the Token Expiry Time per Service Provider feature

### PR DESCRIPTION
**Pull request by GDRDABARERA**

### CODE REVIEW

File: modules/integration/tests-common/admin-clients/src/main/java/org/wso2/identity/integration/common/clients/PropertiesAdminServiceClient.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package org.wso2.identity.integration.common.clients;
+
+/**
+ * Created by wso2dinali on 21/8/17.
```
Comment (kasunbg) [inline modules/integration/tests-common/admin-clients/src/main/java/org/wso2/identity/integration/common/clients/PropertiesAdminServiceClient.java]: unnecessary comment.

File: modules/integration/tests-common/admin-clients/src/main/java/org/wso2/identity/integration/common/clients/PropertiesAdminServiceClient.java [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package org.wso2.identity.integration.common.clients;
```
Comment (kasunbg) [inline modules/integration/tests-common/admin-clients/src/main/java/org/wso2/identity/integration/common/clients/PropertiesAdminServiceClient.java]: Include license header.

File: modules/integration/tests-common/admin-clients/src/main/java/org/wso2/identity/integration/common/clients/PropertiesAdminServiceClient.java [hunk 3]
```diff
@@ -0,0 +1,7 @@
...
+
+/**
+ * Created by wso2dinali on 21/8/17.
+ */
+public class PropertiesAdminServiceClient {
```
Comment (kasunbg) [inline modules/integration/tests-common/admin-clients/src/main/java/org/wso2/identity/integration/common/clients/PropertiesAdminServiceClient.java]: This class seems to be empty.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2TokenTimePerSPTestCase.java [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+                "\"applicationAccessTokenExpireTime\" : 500000,\n" +
+                "\"refreshTokenExpireTime\" : 500000 \n" +
+                "}\n";
+        Boolean success = false;
+
+        try {
+            propAdminServiceClient = new PropertiesAdminServiceClient(backendURL, sessionCookie);
+            propAdminServiceClient.setProperty("_system/config/identity/config/spTokenExpireTime", consumerKey, jsonStringOfTime);
+            success = true;
+        } catch (Exception e) {
```
Comment (kasunbg) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2TokenTimePerSPTestCase.java]: Can we not catch a specific exception?

File: modules/integration/tests-common/admin-clients/src/main/java/org/wso2/identity/integration/common/clients/PropertiesAdminServiceClient.java (hunk 2 above)
Comment (GDRDABARERA) [inline modules/integration/tests-common/admin-clients/src/main/java/org/wso2/identity/integration/common/clients/PropertiesAdminServiceClient.java]: done

File: modules/integration/tests-common/admin-clients/src/main/java/org/wso2/identity/integration/common/clients/PropertiesAdminServiceClient.java [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
+    private static final Log log = LogFactory.getLog(PropertiesAdminServiceClient.class);
+
+    private final String SERVICE_NAME = "PropertiesAdminService";
+    public static final String NAME = "name";
+    public static final String VALUE = "value";
+    public static final String PATH = "path";
+
+    private PropertiesAdminServiceStub stub;
+    private ResourceAdminServiceStub resourceAdminServiceStub;
```
Comment (kasunbg) [inline modules/integration/tests-common/admin-clients/src/main/java/org/wso2/identity/integration/common/clients/PropertiesAdminServiceClient.java]: What is the use of resourceAdminServiceStub field?

File: modules/integration/tests-common/admin-clients/src/main/java/org/wso2/identity/integration/common/clients/PropertiesAdminServiceClient.java [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+        String endPoint = backendURL + SERVICE_NAME;
+        resourceAdminServiceStub = new ResourceAdminServiceStub(endPoint);
+        AuthenticateStub.authenticateStub(userName, password, resourceAdminServiceStub);
+    }
+
+    public void setProperty(String path, String key, String value) throws RegistryException, PropertiesAdminServiceRegistryExceptionException  {
+        try {
+            stub.setProperty(path, key, value);
+        } catch (RemoteException e) {
+            throw new RegistryException("Failed to add property" + key + "to resource at path " + path, e);
```
Comment (kasunbg) [inline modules/integration/tests-common/admin-clients/src/main/java/org/wso2/identity/integration/common/clients/PropertiesAdminServiceClient.java]: Missing spaces after 'property' and and before 'key'.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2TokenTimePerSPTestCase.java [hunk 7]
```diff
@@ -0,0 +1,173 @@
...
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Test case for oauth2 token time per service provider.
+ */
+public class OAuth2TokenTimePerSPTestCase extends OAuth2ServiceAbstractIntegrationTest {
+    private AuthenticatorClient logManger;
```
Comment (kasunbg) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2TokenTimePerSPTestCase.java]: What's the use of logManger?

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2TokenTimePerSPTestCase.java [hunk 8]
```diff
@@ -0,0 +1,173 @@
...
+    private String adminUsername;
+    private String adminPassword;
+    private String accessToken;
+    private String consumerKey;
+    private String consumerSecret;
+    private String sessionDataKeyConsent;
+    private String sessionDataKey;
```
Comment (kasunbg) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2TokenTimePerSPTestCase.java]: What's the use of sessionDataKeyConsent and sessionDataKey?

Make sure to run **Findbugs**. Your class should be shown in green color in IDEA. There should not be any yellow lines.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2TokenTimePerSPTestCase.java (hunk 7 above)
Comment (GDRDABARERA) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2TokenTimePerSPTestCase.java]: In all our integration tests logManger is used to login the IS server as follows,
        logManger.login(isServer.getSuperTenant().getTenantAdmin().getUserName(),
                isServer.getSuperTenant().getTenantAdmin().getPassword(),
                isServer.getInstance().getHosts().get("default"));

**Pull request merged**